xml_converter/src/xml_converter.cpp

```diff
@@ -64,13 +64,38 @@ void write_xml_file(string xml_filepath, map<string, Category>* marker_categorie
     outfile.close();
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// create_proto_categories
+//
+// Fills a set of the names of categories for validation purposes
```

Comment:
Be more descriptive here to avoid confusion over which set. Additionally the purpose of the set created is not mandated by this function in any way so that does not need to be included.

This function might also be better served with a new name to fit its current function, as it is not actually creating a waypoint::Category protobuf object and `proto_categories` is just the name of an argument.